Added parameter service_account_email_instance to gce.py to enable se…

jensonb opened the pull request:
##### ISSUE TYPE

<!--- Pick one below and delete the rest: -->
- Feature Pull Request
##### COMPONENT NAME

<!--- Name of the plugin/module/task -->
- module: GCE
##### ANSIBLE VERSION

<!--- Paste verbatim output from “ansible --version” between quotes below -->

```
ansible 2.1.0.0 (stable-2.1 4c59c0b31d) last updated 2016/06/11 23:14:21 (GMT +000)
  lib/ansible/modules/core: (gce-patch 4b5ca7c40c) last updated 2016/06/12 02:41:17 (GMT +000)
  lib/ansible/modules/extras: (detached HEAD 312540f0e2) last updated 2016/06/11 23:14:32 (GMT +000)
  config file = 
  configured module search path = Default w/o overrides
```
##### SUMMARY

<!--- Describe the change, including rationale and design decisions -->

I needed to be able to set the default service account used by the target instance when I create Google Compute Engine instances via Ansible. I added a parameter, service_account_email_instance, to gce.py so that you can declare a service account to use. When this parameter is not declared, it behaves the same as prior to this commit, passing "default" and allowing Google to assign the default Compute Engine service account.

<!---
If you are fixing an existing issue, please include "Fixes #nnnn" in your commit
message and your description; but you should still explain what the change does.
-->

<!--- Paste verbatim command output below, e.g. before and after your change -->

```

```

…tting the default service account on created instances.

zarnovican committed:
Py2.4: SystemExit in async_wrapper is not an error - compatibility fix

Prior to Python 2.5, SystemExit was a subclass of Exception.
In Py2.4, this is causing extra error output on valid sys.exit(0).

(Toshio) Call sys.exit from inside of the SystemExit exception handler so py2.4 and py2.5+ behaviour matches

jmunhoz committed:
Add Ceph RGW S3 compatibility

Ceph Object Gateway (Ceph RGW) is an object storage interface built on top of
librados to provide applications with a RESTful gateway to Ceph Storage
Clusters:

http://docs.ceph.com/docs/master/radosgw/

This patch adds the required bits to use the RGW S3 RESTful API properly.

Signed-off-by: Javier M. Mellid <jmunhoz@igalia.com>

Commit:
fix #3528 : dns type

gillesgagniard committed:
Initial support for specifying in which subnetwork a gce instance should be created. This is required for non-legacy networks.

gillesgagniard committed:
fix documentation to reflect subnetwork option is new in ansible 2.2

gillesgagniard committed:
better error message if libcloud is not recent enough to use gce subnetworks

stefanha committed:
Stream image data in load_image() to avoid out-of-memory

Reading the entire tar file into memory can result in out-of-memory
conditions such as this traceback:

Traceback (most recent call last):
  File "/tmp/ansible_YELTSu/ansible_module_docker_image.py", line 486, in load_image
    self.client.load_image(image_data)
  File "/usr/local/lib/python2.7/dist-packages/docker/api/image.py", line 147, in load_image
    res = self._post(self._url("/images/load"), data=data)
  ...
  File "/usr/lib/python2.7/httplib.py", line 997, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python2.7/httplib.py", line 848, in _send_output
    msg += message_body
MemoryError

Luckily docker-py's load_image(), which calls requests post(), accepts a
file-like object instead of a string.  Pass in the file object to avoid
reading the full file into memory.  This allows larger tar files to load
succesfully.

Signed-off-by: Stefan Hajnoczi <stefanha@gmail.com>

Commit:
fix unarchive doesn't extract changed tar file #3901

jensonb committed:
Added parameter service_account_email_instance to gce.py to enable setting the default service account on created instances.

gregdek commented:
Thanks @jensonb. To the current maintainers, @erjohnso please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

jensonb commented:
shipit

Commit:
Fix #3941

Commit:
Fixing compile time errors irt (, e => as e, print(), ocat now 0o not 0) exception handling for Python 3  (#3851)

* Fixing compile time errors irt a) exception handling for Python 3 in util, also: b) problem octal usage (fixed) and c) print json_dump -> print(json_dump(xyz) ... et al

* This code was not Python 2.4 compliant. Octal codes and exception handling is now working with Py 2.4, 2.6, & 3.5.

* Fixing formating (or rather reverting an non 2.4 compatible change). Works in compile & runtime checking.

* a) revert to use print sys.stderr not fail_json; b) fixed var name in exception

* Python 3 compatible print (print >>sys.stderr will generate a TypeError - now uses sys.stderr.write instead).

abadger committed:
Finish python3 syntax compilation fixing for core repo

bocharsky-bw committed:
Add mode option to docstring for get_url (#3630)

Add mode option to the list

bcoca committed:
Revert "Add mode option to the list" (#3946)

bcoca committed:
document that get_url also takes file params

has for a long time, but was never documented, mode,group, owner can all be set here

Commit:
Fix #3945 docker_container detach not waiting on container to execute.

Commit:
Remove dns from container create parameters.

nitzmahone committed:
rename ansible_winrm_certificate_expires fact to avoid collision with connection vars

bcoca committed:
updated raw docs for environment and executable

it should now reflect how raw works in combination with
become and environment directives

quater committed:
fixed unknown type <type 'list'> error (#3959)

bcoca committed:
restored incorrectly removed docs, jinja2 override

nitzmahone committed:
StrictMode fixes for win_get_url

adambutler committed:
Fixes incorrect key name protocols -> protocol (#3963)

ewigkeit1204 committed:
Do not copy file if running on check mode

abadger committed:
Fix exception handling in async_wrapper

bcoca committed:
exception ans sys.exit fixes

bcoca committed:
added missing import

readded raise

bcoca committed:
reverted need for module utils imports

XooR committed:
Fix wait_for on python < 2.6 (#3790) (#3973)

jmunhoz committed:
Add proper version_added for rgw option in s3.py

Signed-off-by: Javier M. Mellid <jmunhoz@igalia.com>

jmunhoz committed:
Add s3_url requirement in doc when rgw support is enabled in s3.py

Signed-off-by: Javier M. Mellid <jmunhoz@igalia.com>

mariojmdavid committed:
add project to os_router (#3869)

* add project to os_router

* version_added fix

* correct place of version_added

* remove extraneous else

* * add shade version dependency check
* option project only on router creation

* * fix shade minimal version
* remove pop of option project

* * fix square brackets
* delete router by router_id

Commit:
Fixes #1715 Allow authorized_module accept multi. (#3573)

This will allow the authorized_module to accept options that can be
passed multiple times into ssh options. For instance permitopen.

Commit:
Fix ulimit comparison

Commit:
Fix index

Commit:
Fix compose_version error. Check has compose before checking version.

mgedmin committed:
git: fix change testing when version is HEAD (#3025)

When `version` is not specified, it defaults to "HEAD".  "HEAD" is not a
remote tag, and it's not listed in the output of get_branches(), so we'd
keep repo_updated at the default value (None) and then return early with
changed=True in --check mode, even when before == after.

Fixes #3024.

abadger committed:
Add comment

jctanner committed:
Add a note about using the script module with pseudo-tty's lack of stderr channel (#4000)

Addresses #3610

jensonb committed:
Merge branch 'devel' of https://github.com/ansible/ansible-modules-core into gce-patch

it merges an updated upstream into a topic branch.
